src/components/RefineSearch/RefineSearch.tsx

```diff
@@ -97,24 +97,26 @@ const RefineSearch = ({
       pathname: "/search",
       query: updatedQuery,
     })
-    // refine search dialog closes after url is pushed
     toggleRefine()
   }
 
+  // the two buttons can share a ref because they are never rendered at the same
```

Comment:
Hmm I would think that you would need to update the `refineCancelRef.current` value to get the appropriate DOM element. Perhaps rename `refineOrCancelRef`?

Reply:
Yea I was surprised that it worked as well. I renamed to include `or`.